File: app/src/main/java/mishka/rssreader/RssApplication.java
package mishka.rssreader;

import android.app.Application;

import io.realm.Realm;
import mishka.rssreader.di.component.ApplicationComponent;
import mishka.rssreader.di.component.DaggerApplicationComponent;
import mishka.rssreader.di.module.ApplicationModule;

public class RssApplication extends Application {
    private ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        applicationComponent = DaggerApplicationComponent.builder().applicationModule(new ApplicationModule(this)).build();
        Realm.init(this);
    }

    public ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }
}

File: app/src/main/java/mishka/rssreader/RssApplication.kt
package mishka.rssreader;

import android.app.Application;
import android.content.Context

import mishka.rssreader.di.component.ApplicationComponent;
import mishka.rssreader.di.component.DaggerApplicationComponent;
import mishka.rssreader.di.module.ApplicationModule;
import mishka.rssreader.ui.channelsettings.CHANNEL_URLS_PREF
import mishka.rssreader.ui.channelsettings.CURRENT_CHANNEL_KEY
import mishka.rssreader.ui.channelsettings.CURRENT_CHANNEL_URL_PREF
import mishka.rssreader.ui.channelsettings.ChannelConfigUtils

public class RssApplication: Application() {
    lateinit var applicationComponent: ApplicationComponent
    override fun onCreate() {
        super.onCreate()
        applicationComponent = DaggerApplicationComponent.builder().applicationModule(ApplicationModule(this)).build()
        deleteDatabase("database")




//        ChannelConfigUtils(this).deleteAllChannels()
//        ChannelConfigUtils(this).resetCurrentChannelUrl()
        ChannelConfigUtils(this)
    }
}
